data/transactions/logic/assembler.go

```diff
@@ -1148,8 +1155,8 @@ func (ops *OpStream) resolveLabels() {
 		}
 		// all branch instructions (currently) are opcode byte and 2 offset bytes, and the destination is relative to the next pc as if the branch was a no-op
 		naturalPc := lr.position + 3
-		if dest < naturalPc {
-			ops.errorf("label %v is before reference but only forward jumps are allowed", lr.label)
+		if ops.Version < backBranchEnabledVersion && dest < naturalPc {
+			ops.errorf("label %v is a back reference, back jump support was introduced in TEAL v4", lr.label)
```

Comment:
nit: given that `label` is a string, using `%s` is more appropriate.

Reply:
I'm happy to change, though I think I'd rather change to %#v and get quotes around the label.